Sources/ClientRuntime/Networking/Http/HttpContext.swift

```diff
@@ -108,84 +94,107 @@ public class HttpContextBuilder {
         return self
     }
 
     @discardableResult
-    public func withServiceName(value: String) -> HttpContextBuilder {
-        self.attributes.set(key: serviceName, value: value)
+    public func withMethod(value: HttpMethodType) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.method, value: value)
         return self
     }
 
     @discardableResult
-    public func withDecoder(value: ResponseDecoder) -> HttpContextBuilder {
-        self.attributes.set(key: decoder, value: value)
+    public func withOperation(value: String) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.operation, value: value)
         return self
     }
 
+    /// Sets the partition ID on the context builder.
+    ///
+    /// Requests made with the same partition ID will be grouped together for retry throttling purposes.
+    /// If no partition ID is provided, requests will be partitioned based on the hostname.
+    /// - Parameter value: The partition ID to be set on this builder, or `nil`.
+    /// - Returns: `self`, after the partition ID is set as specified.
     @discardableResult
-    public func withResponse(value: HttpResponse) -> HttpContextBuilder {
-        self.response = value
+    public func withPartitionID(value: String?) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.partitionId, value: value)
         return self
     }
 
     @discardableResult
-    public func withIdempotencyTokenGenerator(value: IdempotencyTokenGenerator) -> HttpContextBuilder {
-        self.attributes.set(key: idempotencyTokenGenerator, value: value)
+    public func withPath(value: String) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.path, value: value)
         return self
     }
 
     @discardableResult
-    public func withLogger(value: LogAgent) -> HttpContextBuilder {
-        self.attributes.set(key: logger, value: value)
+    public func withResponse(value: HttpResponse) -> HttpContextBuilder {
+        self.response = value
         return self
     }
 
-    /// Sets the partition ID on the context builder.
-    ///
-    /// Requests made with the same partition ID will be grouped together for retry throttling purposes.
-    /// If no partition ID is provided, requests will be partitioned based on the hostname.
-    /// - Parameter value: The partition ID to be set on this builder, or `nil`.
-    /// - Returns: `self`, after the partition ID is set as specified.
     @discardableResult
-    public func withPartitionID(value: String?) -> HttpContextBuilder {
-        self.attributes.set(key: partitionID, value: value)
+    public func withServiceName(value: String) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.serviceName, value: value)
         return self
     }
 
     public func build() -> HttpContext {
         return HttpContext(attributes: attributes)
     }
 }
+
+public struct AttributeKeys {
```

Comment:
Also, if you just want to create a namespace in Swift that can't be instantiated, use an `enum` that has no cases, i.e.:
https://github.com/awslabs/aws-sdk-swift/blob/main/Sources/Services/AWSSTS/models/Models.swift#L278

Reply:
I located these attribute keys in `HttpContext` bc this is where they're mainly used & because swift compiler doesn't allow static stored properties in generic type such as `AttributeKey`.

But regarding changing this to enum, that's a good point - I'll change this to enum and push it up.